src/org/mozilla/javascript/NativeJSON.java

```diff
@@ -102,7 +107,29 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
                     }
                 }
 
-                return stringify(cx, scope, value, replacer, space);
+                return stringify(cx, scope, thisObj, value, replacer, space);
+            }
+            
+            case Id_javaConverter: {
+                if (args.length > 0) {
+                    Object value = args[0];
+                    if (value instanceof java.sql.Date) { // Date-Time handling.
+                        value = ((java.sql.Date) value).toLocalDate();
+                    } else if (value instanceof java.sql.Time) {
+                        value = ((java.sql.Time) value).toLocalTime();
+                    } else if (value instanceof java.util.Date) {
+                        value = ((Date) value).toInstant();
+                    } else if (value instanceof Calendar) {
+                        value = ((Calendar)value).toInstant();
+                    } 
+                    if (value instanceof Temporal
+                            || value instanceof UUID) {
```

Comment:
CHECKME: There are a lot of candidates, that may be converted with "toString". Locale, Duration, ...

Reply:
I like how you did this. I was thinking almost the same thing. I still think the default should be to throw TypeErrors, but it should be easy to swap in this implementation. What do you think about another property on the JSON object that is a behavior selector for java objects between TypeError, javaConverter, empty object, or undefined, with this default implementation of javaConverter being provided?

Reply:
The problem is, to distinguish, when use "toString" and when throw a TypeError (or return undefined):
Some brainstorming:
*Idea 1:* IF obj.getClass().startsWith("java.") THEN return obj.toString ELSE throw TypeError
This will cover all java objects from the JRE. But there may be some custom add ons (e.g. joda-time) where toString makes also sense

*Idea 2:* IF isLiveConnect(obj) THEN return obj.toString ELSE throw TypeError
The idea is, to detect, if the object comes from outside (not from the org.mozilla.javascript package). In this case, toString may be the best choice.

*Idea 3:* The user has to provide an implementation to convert LiveConnect objects to JSON.
This is the idea, I mostly like. 
We use a lot of Java-Beans and use Jackson on the java side. It would be great, if we produce the same result like jackson.
And the javaConverter already goes in that direction, but I'm not yet sure if this is the best solution for now.

We use a shared scope for several threads. So if one thread would modify the javaConverter (which should not be possible, as NativeJSON should be sealed), the other thread would see the modification.

What would you think, when we delegate all unconvertable objects to the WrapFactory and it can decide, what to do. 
```java
Object json = cx.getWrapFactory().javaToJson(value, indent, gap)
if (json != null) {
    return json;
}
if (!Undefined.isUndefined(value)) {
    throw ScriptRuntime.typeErrorById("msg.json.cant.serialize", value.getClass().getName());
}
```